README
Build-friendly nukes with realistic effects
Version 1.8
See changelog file for changes

DESCRIPTION:
Realistic nuclear weapons, effects and radiation!

This experimental mod was made to fix several shortcomings of Gbombs-based nukes. Among the issues were no small warheads meant for building (only gigantic early bombs and one-off test articles; no modern real-life deployed models) and somewhat unrealistic and arbitrary effects and power. I forked HBOMBS and started making some new entities, then i decided to go all the way and create an entire Full Realism nuke mod. This is a modification of HBOMBS, it's more of an expansion to it than a stand-alone mod for now, i never modded for Gmod before and didn't know Lua so i started from that. Much credit to it's author, this mod owes a lot to him. 

I made small yet life-size warheads in various yields including a few selectable-yield devices and 4 types of aircraft bombs. They use totally different code from regular nukes for more realistic effects, based on real-life research and scientific models. Adjustments have been made to the scale (1:12), duration and effects of explosions to keep things reasonable for Garry's Mod, yet their retain the right power scaling and effects of real-life, miles ahead of any other nuke mod currently out. The radiation model is also much more advanced than anything else out there, using inspiration from real-life effects of radiation. 

It is in constant work and will be continually updated for better realism.
This is my first mod, so please advise me of any bugs or errors and give me feedback. 
All deserved credit for the original code and models goes to their respective authors, i just improved on their work.

REQUIRED:
Counter Strike Source
Wiremod
SProps
HBOMBS Base Pack
HBOMBS Material Pack
Hazmat Conscripts Playermodels

Bomb power: All bombs now have realistic blast and heat damage ranges from real data, converted to Gmod units and scaled down 1:12. This means that the ranges for the effects of the explosion are realistic (enough), but divided by 12 from real life ranges. A bomb will have more powerful blast effects if it is exploded in the air, but produce no fallout.

Heat damage: The incinerate/ignite code in the bombs was completely redone. Now depending on how close you are from the bomb you will now either be burned slightly, be badly burned and maybe die, be burned to a crisp and die, or even be burned down to your bones. This depends on wether or not you are directly exposed to the heat flash, so hiding behind something will protect you from the flash burns as in real life. This also affects NPC's! Props close to the bomb will catch fire and be charred (material change).

Blast damage: Blast damage now follows realistic ranges and power, props and entities very close to ground zero will simply vaporize. Work in progress.

Dial-a-yield: Some warheads and bombs have a selectable yield option just like in real life, which means you use wire to specify the power of explosion you want.

Radiation model:
Each player has a value that stores the radioactivity dose he has accumulated. Each Rad you have accumulated will deal you 0.001 damage per second. If not exposed to any radiation, this dose will decrease up to 10 units every second. To see what is your current dose, type "/dosimeter" in chat. You will be notified by your geiger counter of the current radiation level. Exposures above 200 rads will start killing you very quickly. Large doses above 1000 rads, even if you survive them, will eventually deal permanent health handicap from massive damage to your DNA and chromosomes. To mute the geiger counter sounds, type /geigersound. all radiation sources (debris, radioactive waste, bombs, prompt radiation; not fallout) now take into consideration the shielding between you and them and how much it attenuates your dose. This means you can now build shielding to protect yourself from radiation. Water and air also act as shielding. Radiation sources now have infinite range, you will always be exposed to some level of radiation. However, this will not harm you, it usually is not even detectable. This reflects how radiation behaves in real life. Props will shield you more than World (buildings, ect), which will shield you more than water, which is better than air. It also considers how much air itself absorbs radiation. 

For reference, 1 Rad is 1000 milirads, and 1 miliRad is 1000 microRads; this means 1 microrad (average background radiation in game is 1.5 microrads/min) is 1/1000000th of a Rad; so don't panic if you see those levels. Microrads = basically nothing, Milirads = not too bad, Rads = dangerous. You can wire the geiger counter entity to multiply gates (x1000 for milirads, x1,000,000 for microrads) to see low levels. 

Radiation Protection:
This is in the works, in the meanwhile you can use Gbombs hazmat suit, it will cut your radiation exposure 80%. There is also two semi-realistic drugs, Iodine and Rad-Away. Iodine will protect you from 25% of radiation for 2 minutes, and Rad-Away will accelerate your accumulated rads elimination rate at the cost of 25HP, allowing you to survive exposure to large doses. You can use props as shielding.

Prompt Radiation:
Prompt radiation is the initial blast of radiation (neutrons and gamma rays) that is produced in the first second the nuke explodes. This is subject to inverse-square law as well as absorption and scattering by the atmosphere, so it's range very rapidly falls off. It is most noticeable on smaller yield bombs, where the physical effects (blast, heat flash) are small enough that the lethal range of radiation actually outranges them. Being not in line-of-sight with the explosion will still give you half the dose you would have had if you were in direct line with it, because the atmosphere is bouncing gamma rays at you from the sky. Being shielded overhead and not in line of sight will only expose you to a fraction of the dose; gamma rays are extremely penetrative and still find a way to reach you, but it's much less. 

Fallout:
Fallout was completely reworked and now decays over time as in real life; on a 2-minute timescale. It starts falling down from the sky and the radiation level peaks about 10 seconds after the explosion, then it starts dropping off quite quickly. A slider will be added for long decay for those who want more realistic consequences. Instead of directly giving you damage, it will add to your radiation dose and you will feel the effects even if you are only exposed for a short while. You are shielded from fallout if you stay inside a building or underground, or if you have a hazmat suit (up to a certain point!). Fallout can expose you to massive doses of radiation, so it is vital that you shelter indoors or wear a hazmat suit if you are close to a bomb that explodes on the ground. Airbursts (bombs exploding in the air) do not produce fallout, the height at which it is considered an airburst depends on the size of the fireball, thus the explosion's power. 

Enhanced Radiation Weapons:
New in 1.4, this is a small tactical warhead that uses an Enhanced Radiation device, commonly known as a Neutron Bomb. This bomb trades blast and heat power for a stronger burst of radiation, making it more lethal against players or NPC's while not having as powerful an explosion as normal bombs. A 1kt ERW has only half the power of a normal 1kt bomb, but the radiation power of a 5kt device; 5kt has the radiation power of a 10kt; and a 10kt that of a 20kt. They tend to be on the smaller scale of nukes, being intended for battlefield use, so 10kt is about the most powerful they come. The REN-ERTAC has 3 modes, 1kt, 5kt and 10kt.

Reactor core:
New in 1.8, this is a nuclear reactor core that emits high levels of radiation. You can power it up for extremely high levels, but beyond 110% power it risks exploding. This is not the complete reactor, this is the EXPOSED CORE, NO SHIELDING. This thing creates enough radiation when operating at full power to instantly kill you, you must building shielding around it just like a real reactor. If you power it up too high and it explodes, you are in BIG trouble. It creates a massive (non-nuclear) explosion and shoots radioactive fallout and debris everywhere. And i mean literal debris, extremely radioactive rocks that you have to somehow remove from the area.  But the core is still there, destroyed but still shooting out insanely lethal amounts of radiation. You now have to explain to your boss how an RBMK reactor explodes.


REFERENCES:
Nukemap by Alex Wellerstein nuclearsecrecy.com/nukemap/
"The Effects of Nuclear Weapons" by S. Glasstone and P. J. Dolan

Note:
kt = kilotons (1000 tons of TNT)
mt = megatons (1000 kilotons)

In this mod you will find the following:
REN-1 warhead (1 kiloton) (visuals of Davy Crockett, no flash)
REN-5 warhead (5 kilotons) (visuals of Davy Crockett, with flash)
REN-10 warhead (10 kilotons) (visuals of Artillery shell)
REN-20 warhead (20 kilotons) (visuals of Fat Man)
REN-50 warhead (50 kilotons) (visuals of Ivy King)
REN-100 warhead (100 kilotons) (visuals of Gadget)
REN-250 warhead (250 kilotons) (visuals of Little Boy)
REN-500 warhead (500 kilotons) (visuals of Ivy Mike)
REN-1000 warhead (1 megaton) (visuals of Castle Bravo)
REN-5000 warhead (5 megatons) (visuals of Tsar Bomba)
REN-TAC warhead - Tactical warhead with selectable yield, Modes are 0=1kt, 1=5kt, 2=10kt, 3=20kt, 4=50kt, default (0)=1kt
REN-STRAT warhead - Strategic warhead with selectable yield, Modes are 0=100kt, 1=250kt, 2=500kt, 3=1mt, default (0)=100kt
REN-ERTAC warhead - Enhanced Radiation tactical warhead (neutron bomb), Modes are 0=1kt, 1=5kt, 2=10kt, default (0)=1kt
Mk12 Bomb - Tactical bomb, same thing as REN-10
B61 Mod 7 bomb - Strategic bomb with selectable yield, Modes are 1=20kt, 2=50kt, 3=100kt, 4=250kt; default (0)=100kt
B61 Mod 12 bomb - Tactical bomb with selectable yield, Modes are 1=1kt, 2=5kt, 3=10kt, 4=20kt, 5=50kt, default (0)=50kt
B83 bomb - Strategic bomb with selectable yield, Modes are 1=100kt, 2=250kt, 3=500kt, 4=1mt, default (0)=1mt
Dirty bomb - Conventional 500lb bomb but with radioactive material inside, creates lots of fallout
Nuclear waste barrel - A highly radioactive barrel of nuclear waste
Radioactive debris - Spicy rocks from an RBMK reactor
Nuclea reactor core - Highly radioactive core that can be powered up to massive levels of radiation, and potentially explode with catastrophic consequences

FAQ
Q: Why are the warheads so tiny?
A: Because that's about how big they are in real life when you remove the bomb or missile warhead casing that holds them. They are meant to be easy to fit and weld into a contraption you make, such as a missile.

Q: Why are the bombs so small?
A: Because that's about how big they are in real life. You don't need a bomb the size of a freaking 18-wheeler trailer to reach megatons of power, it's not 1952 anymore.

Q: Why are they all just boring cylinders?
That's the point. They're small, handy little things you can fit on to anything you build. The bombs are there if you want something more exciting-looking.

Q: Why are the yields you give so scaled down? IE: why does the Tsar Bomba equivalent only say it's 5 megatons?
A: The Tsar Bomba in, say GBombs, is about visually equivalent to the real Little Boy. In effects, it's about really equivalent to just a 8-10 kiloton nuke in real life. The Tsar Bomba's power, 57 megatons (57,000 kilotons) is just mindboggling, like the wrath of god. The biggest GMod map i can find is about 860 meters diagonally, it would fit entirely within the fireball of a real 400 kiloton explosion. This is how big these things are. With the 1:12 scale i'm now using, 5 megatons is about as big as you can get without everything on the map being instantly vaporized (just barely). The new realistic power calculation give the bombs good and realistic power for our use.

Q: The bombs are overpowered!
A: Overpowered... realistic... nuclear... weapons... Think about that for a second. If you think there's something that could be adjusted for better effects and playability, tell me about it, but being overpowered is kind of the entire thing with nukes. 

Q: Can you make the Tsar Bomba/This specific bomb/Some fantasy bomb?
A: No. The point of this mod is realism and to leave it up to you to make anything you want instead of making specific weapons.

Code and more info on Github: https://github.com/snow-frog/EnduringStockpile/


changelog
1.8 CHANGELOG:
- Nuclear reactor core explodes
- HUD replaces notification messages for geiger counter
- Updates to radiation halflife settings
- More convars
- Changed Dirty Bomb model
- Changed and included Radioactive Waste Barrel


1.7 CHANGELOG:
- Updated radiation models
- High radiation levels can now fry electronics
- Shielding now applies to electronics
- New icons
- Added Nuclear reactor core


1.6 CHANGELOG:
All nuclear bombs now emit a tiny amount of radiation (microrads scale).

HUGE CHANGE: RADIATION SHIELDING
This is extremely important for realism, now all radiation sources (debris, radioactive waste, bombs, prompt radiation; not fallout) now take into consideration the shielding between you and them. For simplicity, i only consider gamma radiation. This is not perfect, but it will consider the thickness of whatever is between you and it and find out by how much it attenuates your dose. This means you can now build shielding to protect yourself from radiation. Water and air also act as shielding. This also means that radiation sources now have infinite range, you will always be exposed to some level of radiation. However, this will not harm you, it usually is not even detectable; it would take one real week to lose 1 HP from 100 milirads/min. This reflects how radiation behaves in real life. Props will shield you more than World (buildings, ect), which will shield you more than water, which is better than air. I not only consider inverse-square law for sources, but also how much air itself absorbs radiation. All things considered is a HUGE improvement.
 
 For reference, 1 Rad is 1000 milirads, and 1 miliRad is 1000 microRads; this means 1 microrad (average background radiation in game is 1.5 microrads/min) is 1/1000000th of a Rad; so don't panic if you see those levels. Microrads = basically nothing, Milirads = not too bad, Rads = dangerous. You can wire the geiger counter entity to multiply gates (x1000 for milirads, x1,000,000 for microrads) to see low levels. 

Other changes, you now spawn with no dosimeter, so you won't know your exposure level nor radiation level whitout getting one. Just press E on it to equip it.

Added two anti-radiation drugs, not 100% realistic but useful for gameplay. Iodine will cut down your Rad absorption by 25% for 2 minutes. Does not stack. Rad-Away, will increase your Rad elimination rate for 2 minutes at the cost of 25HP. Does not stack.

CHANGES:
- Updated radiation models
- Added Dosimeter
- Added Iodine
- Added Rad-Away
- Added Radioactive debris
- Added Low level radiation detection
- Added Radiation shielding
- Added Radioactive half-life decay
- All bombs now emit extremely small levels of radiation


1.5 CHANGELOG:

Picked it up again and did some internally significant tweaks for radiation to be more realistic. Now radiation does a little less damage on the lower ends but is absolutely DEADLY at higher doses (which you can easily get in seconds if you fuck around in radioactive craters and immediate fallout). It now does a little less immediate damage to health, but it also now does PERMANENT damage, so even if you survive a big dose you will have a health penalty until you die because your chromosomes are damaged. This also happens for low-level irradiation over long periods of time.

I'll add icons when i can get VTF plugins to work on Linux. Modding on linux is hell on earth, absolutely no tool works. I can't upload this to the workshop yet for this reason...

CHANGES:
- Updated radiation models, lots of tweaks
- Optimized radiation code (autorun scripts)
- Permanent chromosonal damage to health for large accumulated doses
- Geiger counter now accurate and can emit sound if wired on
- You are no longer protected from the crater if you are underground, there's supposed to be a crater there anyway
- Added radioactive waste barrel
- Added Dirty bomb

1.4 CHANGELOG:
- 4.0 is now 1.4


gmod_nuke_calc.py
import math

def roundup(x):
    return int(math.ceil(x / 100.0)) * 100


data = """1kt
Fireball radius: 80 m 
Air blast radius (200 psi): 80 m 
Air blast radius (20 psi): 220 m 
Air blast radius (10 psi): 310 m 
Air blast radius (5 psi): 460 m 
Air blast radius (1.5 psi): 0.98 km 
Radiation radius (5000 rem): 0.51 km 
Radiation radius (1000 rem): 0.73 km 
Radiation radius (500 rem): 0.84 km
Thermal radiation radius (100 cal/cm2): 80 m 
Thermal radiation radius (35 cal/cm2): 230 m 
Thermal radiation radius (3rd degree burns): 0.5 km
Thermal radiation radius (2nd degree burns (50%)): 0.66 km
Thermal radiation radius (1st degree burns (50%)): 0.94 km 

1kt air best
Fireball radius: 60 m 
Air blast radius (200 psi): 80 m 
Air blast radius (20 psi): 280 m 
Air blast radius (10 psi): 440 m
Air blast radius (5 psi): 0.7 km
Air blast radius (1.5 psi): 1.62 km 
Radiation radius (5000 rem): 0.51 km
Radiation radius (1000 rem): 0.73 km
Radiation radius (500 rem): 0.84 km 
Thermal radiation radius (100 cal/cm2): 160 m 
Thermal radiation radius (35 cal/cm2): 270 m 
Thermal radiation radius (3rd degree burns): 0.6 km
Thermal radiation radius (2nd degree burns (50%)): 0.79 km
Thermal radiation radius (1st degree burns (50%)): 1.12 km

10kt
Fireball radius: 200 m 
Air blast radius (200 psi): 170 m 
Air blast radius (20 psi): 470 m 
Air blast radius (10 psi): 0.67 km 
Air blast radius (5 psi): 0.99 km 
Air blast radius (1.5 psi): 2.11 km 
Radiation radius (5000 rem): 0.84 km 
Radiation radius (1000 rem): 1.12 km 
Radiation radius (500 rem): 1.25 km 
Thermal radiation radius (100 cal/cm2): 420 m 
Thermal radiation radius (35 cal/cm2): 0.71 km 
Thermal radiation radius (3rd degree burns): 1.41 km
Thermal radiation radius (2nd degree burns (50%)): 1.85 km 
Thermal radiation radius (1st degree burns (50%)): 2.58 km

10kt air best
Fireball radius: 150 m 
Air blast radius (200 psi): 180 m 
Air blast radius (20 psi): 0.61 km
Air blast radius (10 psi): 0.94 km
Air blast radius (5 psi): 1.5 km
Air blast radius (1.5 psi): 3.49 km
Radiation radius (5000 rem): 0.84 km
Radiation radius (1000 rem): 1.12 km
Radiation radius (500 rem): 1.25 km
Thermal radiation radius (100 cal/cm2): 0.5 km 
Thermal radiation radius (35 cal/cm2): 0.84 km
Thermal radiation radius (3rd degree burns): 1.67 km
Thermal radiation radius (2nd degree burns (50%)): 2.2 km
Thermal radiation radius (1st degree burns (50%)): 3.06 km

20kt
Fireball radius: 260 m 
Air blast radius (200 psi): 220 m 
Air blast radius (20 psi): 0.59 km
Air blast radius (10 psi): 0.85 km 
Air blast radius (5 psi): 1.24 km 
Air blast radius (1.5 psi): 2.65 km 
Radiation radius (5000 rem): 0.97 km 
Radiation radius (1000 rem): 1.27 km 
Radiation radius (500 rem): 1.41 km 
Thermal radiation radius (100 cal/cm2): 0.59 km 
Thermal radiation radius (35 cal/cm2): 0.99 km 
Thermal radiation radius (3rd degree burns): 1.91 km 
Thermal radiation radius (2nd degree burns (50%)): 2.52 km 
Thermal radiation radius (1st degree burns (50%)): 3.52 km

20kt air best
Fireball radius: 200 m 
Air blast radius (200 psi): 220 m 
Air blast radius (20 psi): 0.77 km 
Air blast radius (10 psi): 1.19 km
Air blast radius (5 psi): 1.89 km 
Air blast radius (1.5 psi): 4.39 km
Radiation radius (5000 rem): 0.97 km 
Radiation radius (1000 rem): 1.27 km 
Radiation radius (500 rem): 1.41 km 
Thermal radiation radius (100 cal/cm2): 0.71 km 
Thermal radiation radius (35 cal/cm2): 1.18 km 
Thermal radiation radius (3rd degree burns): 2.27 km
Thermal radiation radius (2nd degree burns (50%)): 2.99 km 
Thermal radiation radius (1st degree burns (50%)): 4.16 km

50kt
Fireball radius: 380 m 
Air blast radius (200 psi): 300 m
Air blast radius (20 psi): 0.8 km
Air blast radius (10 psi): 1.45 km
Air blast radius (5 psi): 1.69 km
Air blast radius (1.5 psi): 3.6 km
Radiation radius (5000 rem): 1.16 km
Radiation radius (1000 rem): 1.49 km 
Radiation radius (500 rem): 1.64 km 
Thermal radiation radius (100 cal/cm2): 0.93 km 
Thermal radiation radius (35 cal/cm2): 1.54 km 
Thermal radiation radius (3rd degree burns): 2.87 km
Thermal radiation radius (2nd degree burns (50%)): 3.79 km 
Thermal radiation radius (1st degree burns (50%)): 5.26 km

50kt air best
Fireball radius: 290 m 
Air blast radius (200 psi): 300 m 
Air blast radius (20 psi): 1.04 km 
Air blast radius (10 psi): 1.61 km
Air blast radius (5 psi): 2.56 km 
Air blast radius (1.5 psi): 5.96 km 
Radiation radius (5000 rem): 1.16 km 
Radiation radius (1000 rem): 1.49 km 
Radiation radius (500 rem): 1.64 km 
Thermal radiation radius (100 cal/cm2): 1.1 km 
Thermal radiation radius (35 cal/cm2): 1.83 km 
Thermal radiation radius (3rd degree burns): 3.4 km 
Thermal radiation radius (2nd degree burns (50%)): 4.48 km
Thermal radiation radius (1st degree burns (50%)): 6.21 km

100kt
Fireball radius: 500 m 
Air blast radius (200 psi): 370 m 
Air blast radius (20 psi): 1.01 km 
Air blast radius (10 psi): 1.45 km 
Air blast radius (5 psi): 2.12 km 
Air blast radius (1.5 psi): 4.54 km 
Radiation radius (5000 rem): 1.33 km 
Radiation radius (1000 rem): 1.67 km 
Radiation radius (500 rem): 1.82 km 
Thermal radiation radius (100 cal/cm2): 1.3 km 
Thermal radiation radius (35 cal/cm2): 2.14 km 
Thermal radiation radius (3rd degree burns): 3.9 km 
Thermal radiation radius (2nd degree burns (50%)): 5.14 km 
Thermal radiation radius (1st degree burns (50%)): 7.1 km

100kt air best
Fireball radius: 380 m 
Air blast radius (200 psi): 380 m
Air blast radius (20 psi): 1.31 km
Air blast radius (10 psi): 2.03 km 
Air blast radius (5 psi): 3.23 km 
Air blast radius (1.5 psi): 7.51 km 
Radiation radius (5000 rem): 1.33 km 
Radiation radius (1000 rem): 1.67 km 
Radiation radius (500 rem): 1.82 km 
Thermal radiation radius (100 cal/cm2): 1.54 km
Thermal radiation radius (35 cal/cm2): 2.54 km 
Thermal radiation radius (3rd degree burns): 4.62 km 
Thermal radiation radius (2nd degree burns (50%)): 6.07 km
Thermal radiation radius (1st degree burns (50%)): 8.37 km 

250kt
Fireball radius: 0.72 km 
Air blast radius (200 psi): 0.51 km
Air blast radius (20 psi): 1.37 km 
Air blast radius (10 psi): 1.97 km 
Air blast radius (5 psi): 2.88 km 
Air blast radius (1.5 psi): 6.16 km 
Radiation radius (5000 rem): 1.56 km 
Radiation radius (1000 rem): 1.93 km 
Radiation radius (500 rem): 2.09 km 
Thermal radiation radius (100 cal/cm2): 2.01 km 
Thermal radiation radius (35 cal/cm2): 3.31 km 
Thermal radiation radius (3rd degree burns): 5.84 km 
Thermal radiation radius (2nd degree burns (50%)): 7.68 km 
Thermal radiation radius (1st degree burns (50%)): 10.5 km

250kt air best
Fireball radius: 0.55 km 
Air blast radius (200 psi): 0.52 km
Air blast radius (20 psi): 1.78 km 
Air blast radius (10 psi): 2.76 km 
Air blast radius (5 psi): 4.38 km 
Air blast radius (1.5 psi): 10.2 km 
Radiation radius (5000 rem): 1.56 km
Radiation radius (1000 rem): 1.93 km 
Radiation radius (500 rem): 2.09 km 
Thermal radiation radius (100 cal/cm2): 2.39 km 
Thermal radiation radius (35 cal/cm2): 3.92 km 
Thermal radiation radius (3rd degree burns): 6.89 km
Thermal radiation radius (2nd degree burns (50%)): 9.06 km 
Thermal radiation radius (1st degree burns (50%)): 12.4 km

500kt
Fireball radius: 0.95 km 
Air blast radius (200 psi): 0.64 km 
Air blast radius (20 psi): 1.73 km 
Air blast radius (10 psi): 2.48 km 
Air blast radius (5 psi): 3.63 km 
Air blast radius (1.5 psi): 7.76 km 
Radiation radius (5000 rem): 1.75 km 
Radiation radius (1000 rem): 2.13 km 
Radiation radius (500 rem): 2.29 km 
Thermal radiation radius (100 cal/cm2): 2.8 km 
Thermal radiation radius (35 cal/cm2): 4.59 km 
Thermal radiation radius (3rd degree burns): 7.91 km 
Thermal radiation radius (2nd degree burns (50%)): 10.4 km 
Thermal radiation radius (1st degree burns (50%)): 14.1 km 

500kt air best
Fireball radius: 0.73 km 
Air blast radius (200 psi): 0.65 km 
Air blast radius (20 psi): 2.24 km 
Air blast radius (10 psi): 3.48 km 
Air blast radius (5 psi): 5.52 km 
Air blast radius (1.5 psi): 12.8 km 
Radiation radius (5000 rem): 1.75 km 
Radiation radius (1000 rem): 2.13 km 
Radiation radius (500 rem): 2.29 km 
Thermal radiation radius (100 cal/cm2): 3.31 km 
Thermal radiation radius (35 cal/cm2): 5.43 km 
Thermal radiation radius (3rd degree burns): 9.32 km 
Thermal radiation radius (2nd degree burns (50%)): 12.2 km 
Thermal radiation radius (1st degree burns (50%)): 16.6 km

1mt
Fireball radius: 1.26 km 
Air blast radius (200 psi): 0.8 km 
Air blast radius (20 psi): 2.18 km 
Air blast radius (10 psi): 3.12 km 
Air blast radius (5 psi): 4.58 km 
Air blast radius (1.5 psi): 9.78 km 
Radiation radius (5000 rem): 1.94 km 
Radiation radius (1000 rem): 2.33 km 
Radiation radius (500 rem): 2.5 km 
Thermal radiation radius (100 cal/cm2): 3.88 km 
Thermal radiation radius (35 cal/cm2): 6.35 km 
Thermal radiation radius (3rd degree burns): 10.7 km 
Thermal radiation radius (2nd degree burns (50%)): 14 km 
Thermal radiation radius (1st degree burns (50%)): 19 km 

1mt air best
Fireball radius: 0.97 km 
Air blast radius (200 psi): 0.82 km 
Air blast radius (20 psi): 2.82 km 
Air blast radius (10 psi): 4.38 km 
Air blast radius (5 psi): 6.96 km 
Air blast radius (1.5 psi): 16.2 km 
Radiation radius (5000 rem): 1.94 km 
Radiation radius (1000 rem): 2.33 km 
Radiation radius (500 rem): 2.5 km 
Thermal radiation radius (100 cal/cm2): 4.59 km 
Thermal radiation radius (35 cal/cm2): 7.49 km 
Thermal radiation radius (3rd degree burns): 12.6 km 
Thermal radiation radius (2nd degree burns (50%)): 16.5 km 
Thermal radiation radius (1st degree burns (50%)): 22.3 km

5mt
Fireball radius: 2.39 km 
Air blast radius (200 psi): 1.38 km 
Air blast radius (20 psi): 3.72 km 
Air blast radius (10 psi): 5.34 km 
Air blast radius (5 psi): 7.83 km 
Air blast radius (1.5 psi): 16.7 km 
Radiation radius (5000 rem): 2.46 km 
Radiation radius (1000 rem): 2.88 km 
Radiation radius (500 rem): 3.05 km 
Thermal radiation radius (100 cal/cm2): 8.23 km 
Thermal radiation radius (35 cal/cm2): 13.3 km 
Thermal radiation radius (3rd degree burns): 21.3 km 
Thermal radiation radius (2nd degree burns (50%)): 28 km 
Thermal radiation radius (1st degree burns (50%)): 37.5 km

5mt air best
Fireball radius: 1.84 km 
Air blast radius (200 psi): 1.41 km 
Air blast radius (20 psi): 4.82 km 
Air blast radius (10 psi): 7.49 km
Air blast radius (5 psi): 11.9 km 
Air blast radius (1.5 psi): 27.7 km
Radiation radius (5000 rem): 2.46 km 
Radiation radius (1000 rem): 2.88 km 
Radiation radius (500 rem): 3.05 km 
Thermal radiation radius (100 cal/cm2): 9.69 km 
Thermal radiation radius (35 cal/cm2): 15.7 km 
Thermal radiation radius (3rd degree burns): 25 km
Thermal radiation radius (2nd degree burns (50%)): 32.8 km
Thermal radiation radius (1st degree burns (50%)): 43.8 km"""

def gr_to_sr(gr, height):
    return math.sqrt(math.pow(gr,2) + math.pow(height,2))

testdata = {}
exdata = {
    'fireball': 0,
    '200psi': 0,
    '20psi': 0,
    '10psi': 0,
    '5psi': 0,
    '1.5psi': 0,
    '5krem': 0,
    '1krem': 0,
    '500rem': 0,
    'burn5': 0,
    'burn4': 0,
    'burn3': 0,
    'burn2': 0,
    'burn1': 0
    }

#parse nukemap data
x=data.split('\n\n')
powers = []
c=0
print ''
print 'Real data'
print 'Ground burst'
for i in x:
    print ''
    y = i.split('\n')
    power = y.pop(0)
    testdata[power] = {}
    print power
    powers.append(power)
    for j in range(len(y)):
        try:
            d = float(y[j].split(': ')[1].split(' km')[0])*1000
        except ValueError:
            d = float(y[j].split(': ')[1].split(' m')[0])
        effect = y[j].split(': ')[0]
        
        if effect.startswith('Fireball'):
            entry = 'fireball'
        elif effect.startswith('Air blast radius (200'):
            entry = '200psi'
        elif effect.startswith('Air blast radius (20'):
            entry = '20psi'
        elif effect.startswith('Air blast radius (10'):
            entry = '10psi'
        elif effect.startswith('Air blast radius (5'):
            entry = '5psi'
        elif effect.startswith('Air blast radius (1.5'):
            entry = '1.5psi'
        elif effect.startswith('Radiation radius (5000'):
            entry = '5000rem'
        elif effect.startswith('Radiation radius (1000'):
            entry = '1000rem'
        elif effect.startswith('Radiation radius (500'):
            entry = '500rem'
        elif effect.startswith('Thermal radiation radius (100'):
            entry = 'burn5'
        elif effect.startswith('Thermal radiation radius (35'):
            entry = 'burn4'
        elif effect.startswith('Thermal radiation radius (3rd'):
            entry = 'burn3'
        elif effect.startswith('Thermal radiation radius (2nd'):
            entry = 'burn2'
        elif effect.startswith('Thermal radiation radius (1st'):
            entry = 'burn1'

        #print effect
        print entry,str(int(d))+'m'
        testdata[power][entry] = d

#convert 
scalefactor = 12
uconv = float(52.493)
scale = float(1)/scalefactor

print ''
print ''
print 'Gmod conversion'
print '1:'+str(scalefactor),'scale'
print 'Gmod standard units'
for power in powers:
    print ''
    print power
    print 'fireball',roundup(testdata[power]['fireball'] * uconv * scale)
    print '200psi',roundup(testdata[power]['200psi'] * uconv * scale)
    print '10psi',roundup(testdata[power]['10psi'] * uconv * scale)
    print '5psi',roundup(testdata[power]['5psi'] * uconv * scale)
    print '1.5psi',roundup(testdata[power]['1.5psi'] * uconv * scale)
    print '5000rem',roundup(testdata[power]['5000rem'] * uconv * scale)
    print '1000rem',roundup(testdata[power]['1000rem'] * uconv * scale)
    print '500rem',roundup(testdata[power]['500rem'] * uconv * scale)
    print 'burn5',roundup(testdata[power]['burn5'] * uconv * scale)
    print 'burn4',roundup(testdata[power]['burn4'] * uconv * scale)
    print 'burn3',roundup(testdata[power]['burn3'] * uconv * scale)
    print 'burn2',roundup(testdata[power]['burn2'] * uconv * scale)
    print 'burn1',roundup(testdata[power]['burn1'] * uconv * scale)
